smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/WaiterGenerator.java

```diff
@@ -0,0 +1,223 @@
+    }
+
+    private void generateSuccessMatcher(Matcher.SuccessMember member, AcceptorState state) {
+        writer.write("return $L", getWaiterResult(state));
+    }
+
+    private void generateErrorMatcher(String accessor, Matcher.ErrorTypeMember member, AcceptorState state) {
+        writer.openBlock("if ($L.name && $L.name == $S) {", "}", accessor, accessor,
+                member.getValue(), () -> {
+            writer.write("return $L", getWaiterResult(state));
+        });
+    }
+
+    private void generatePathMatcher(String accessor, PathMatcher pathMatcher, AcceptorState state) {
+        writer.openBlock("try {", "} catch (e) {}", () -> {
+            JmespathExpression expression = JmespathExpression.parse(pathMatcher.getPath());
+            TypeScriptJmesPathVisitor expressionVisitor = new TypeScriptJmesPathVisitor(writer, accessor, expression);
+            String expectedState = "{ state: '" + state.toString() + "'}";
+            expressionVisitor.run();
+
+            switch (pathMatcher.getComparator()) {
+                case ALL_STRING_EQUALS:
+                    expressionVisitor.writeAllStringEqualsExpectation(pathMatcher.getExpected(), expectedState);
+                    break;
+                case ANY_STRING_EQUALS:
+                    expressionVisitor.writeAnyStringEqualsExpectation(pathMatcher.getExpected(), expectedState);
+                    break;
+                case STRING_EQUALS:
+                    expressionVisitor.writeStringExpectation(pathMatcher.getExpected(), expectedState);
+                    break;
+                case BOOLEAN_EQUALS:
+                    expressionVisitor.writeBooleanExpectation(pathMatcher.getExpected(), expectedState);
+                    break;
+                default:
+                    throw new CodegenException("Invalid Matcher Comparator");
+            }
+        });
+    }
+
+    private void writeAcceptors(String accessor, boolean isException) {
+        waiter.getAcceptors().forEach((Acceptor acceptor) -> {
+            switch (acceptor.getMatcher().getMemberName()) {
+               case "success":
+                   Matcher.SuccessMember successMember = (Matcher.SuccessMember) acceptor.getMatcher();
+                   if (successMember.getValue() != isException) {
+                       generateSuccessMatcher(successMember, acceptor.getState());
+                   }
+                   break;
+               case "errorType":
+                   if (isException) {
+                       generateErrorMatcher(accessor, (Matcher.ErrorTypeMember) acceptor.getMatcher(),
+                               acceptor.getState());
+                   }
+                   break;
+               case "inputOutput":
+                   if (!isException) {
+                       Matcher.InputOutputMember member = (Matcher.InputOutputMember) acceptor.getMatcher();
+                       generatePathMatcher(accessor, member.getValue(), acceptor.getState());
```

Comment:
Is this right? The variable the path matcher operates on has "input" and "output" top-level variables, and expression are written with that in mind (i.e., the expressions themselves contains stuff like "output.sprockets"). I see something here for "input", but not "output".

Reply:
output is the accessor. usually just "result"